osu.Game/Screens/Ranking/Expanded/Accuracy/AccuracyCircle.cs

```diff
@@ -191,8 +193,8 @@ private void load()
                     Padding = new MarginPadding { Vertical = -15, Horizontal = -20 },
                     Children = new[]
                     {
-                        new RankBadge(1f, ScoreRank.X),
-                        new RankBadge(0.95f, ScoreRank.S),
+                        new RankBadge(1f, score.Mods.Any(m => m is ModHidden || m is ModFlashlight) ? ScoreRank.XH : ScoreRank.X),
+                        new RankBadge(0.95f, score.Mods.Any(m => m is ModHidden || m is ModFlashlight) ? ScoreRank.SH : ScoreRank.S),
```

Comment:
Use `Mod.AdjustRank(ScoreRank)` instead.

Reply:
Not sure if I did this right. I took the other usage of `AdjustRank`, but it seems error prone when there are multiple rank changing mods. Doesn't really matter for now since hidden and flashlight have the same behavior.
```c#
                badges = new Container<RankBadge>
                {
                    Name = "Rank badges",
                    RelativeSizeAxes = Axes.Both,
                    Padding = new MarginPadding { Vertical = -15, Horizontal = -20 },
                    Children = new[]
                    {
                        new RankBadge(1f, getRank(ScoreRank.X)),
                        new RankBadge(0.95f, getRank(ScoreRank.S)),
                        new RankBadge(0.9f, getRank(ScoreRank.A)),
                        new RankBadge(0.8f, getRank(ScoreRank.B)),
                        new RankBadge(0.7f, getRank(ScoreRank.C)),
                        new RankBadge(0.35f, getRank(ScoreRank.D)),
                    }
                },
                rankText = new RankText(score.Rank)
            };
        }

        private ScoreRank getRank(ScoreRank rank)
        {
            foreach (var mod in score.Mods.OfType<IApplicableToScoreProcessor>())
                rank = mod.AdjustRank(rank, score.Accuracy);

            return rank;
        }
```

*Flashlight doesn't have `AdjustRank` implemented right now, so should I copy the code from hidden and put it in this PR?*

Reply:
Sure, I think it's an innocent enough change to be bundled in here.